Build script fails on NixOS

### I don't have very much knowledge on topic of coding and such... so I'll post my log in hopes that I will receive some support in some way.

```
gcc -MT toplev.o -MMD -MP -MF .d/toplev.Td -g -std=gnu11 -Werror-implicit-function-declaration -DOLD_COMPILER  -I. -I. -c toplev.c
toplev.c: In function ‘fatal_insn’:
toplev.c:1049:5: error: format not a string literal and no format arguments [-Werror=format-security]
 1049 |     error(message);
      |     ^~~~~
toplev.c: In function ‘v_message_with_decl’:
toplev.c:1235:26: warning: ‘%ld’ directive writing between 1 and 20 bytes into a region of size 5 [-Wformat-overflow=]
 1235 |         sprintf(fmt, "%%.%lds", width);
      |                          ^~~
toplev.c:1235:22: note: directive argument in the range [-9223372036854775808, 255]
 1235 |         sprintf(fmt, "%%.%lds", width);
      |                      ^~~~~~~~~
In file included from /nix/store/07kmxfmj8i8h76zzz9vw9zfvxsarqr5d-glibc-2.34-210-dev/include/stdio.h:888,
                 from thumb.h:1078,
                 from telf.h:25,
                 from config.h:12,
                 from toplev.c:26:
/nix/store/07kmxfmj8i8h76zzz9vw9zfvxsarqr5d-glibc-2.34-210-dev/include/bits/stdio2.h:38:10: note: ‘__builtin___sprintf_chk’ output between 5 and 24 bytes into a destination of size 7
   38 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   39 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   40 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make: *** [Makefile:81: toplev.o] Error 1
make: Leaving directory '/home/tryzen/Code/agbcc/gcc'
```

--- a/gcc/c-typeck.c
+++ b/gcc/c-typeck.c
@@ -3133,7 +3133,7 @@ build_unary_op (code, xarg, noconvert)
       return fold (build1 (code, argtype, arg));
     }
 
-  error (errstring);
+  error ("%s", errstring);
   return error_mark_node;
 }
 

--- a/gcc/real.c
+++ b/gcc/real.c
@@ -4042,8 +4042,6 @@ mtherr (name, code)
      char *name;
      int code;
 {
-  char errstr[80];
-
   /* The string passed by the calling program is supposed to be the
      name of the function in which the error occurred.
      The code argument selects which error message string will be printed.  */
@@ -4066,9 +4064,8 @@ mtherr (name, code)
     name = "modulus";
   else if (strcmp (name, "esqrt") == 0)
     name = "square root";
-  sprintf (errstr, "%s during real %s", ermsg[code], name);
   if (extra_warnings)
-    warning (errstr);
+    warning ("%s during real %s", ermsg[code], name);
   /* Set global error message word */
   merror = code + 1;
 }

--- a/gcc/toplev.c
+++ b/gcc/toplev.c
@@ -1061,7 +1061,7 @@ fatal_io_error(char *name)
 void
 fatal_insn(char *message, rtx insn)
 {
-    error(message);
+    error("%s", message);
     debug_rtx(insn);
     if (asm_out_file)
         fflush(asm_out_file);

--- a/gcc_arm/c-common.c
+++ b/gcc_arm/c-common.c
@@ -1587,7 +1587,7 @@ check_format_info (info, params)
 	      ++format_chars;
 	      if (params == 0)
 		{
-		  warning (tfaff);
+		  warning ("%s", tfaff);
 		  return;
 		}
 	      if (info->first_arg_num != 0)
@@ -1630,7 +1630,7 @@ check_format_info (info, params)
 		      ++format_chars;
 		      if (params == 0)
 		        {
-			  warning (tfaff);
+			  warning ("%s", tfaff);
 			  return;
 			}
 		      cur_param = TREE_VALUE (params);
@@ -1825,7 +1825,7 @@ check_format_info (info, params)
 	continue;
       if (params == 0)
 	{
-	  warning (tfaff);
+	  warning ("%s", tfaff);
 	  return;
 	}
       cur_param = TREE_VALUE (params);

--- a/gcc_arm/c-typeck.c
+++ b/gcc_arm/c-typeck.c
@@ -3147,7 +3147,7 @@ build_unary_op (code, xarg, noconvert)
       return fold (build1 (code, argtype, arg));
     }
 
-  error (errstring);
+  error ("%s", errstring);
   return error_mark_node;
 }
 

--- a/gcc_arm/cccp.c
+++ b/gcc_arm/cccp.c
@@ -8676,7 +8676,7 @@ macroexpand (hp, op)
       else
 	parse_error = macarg (NULL_PTR, 0);
       if (parse_error) {
-	error_with_line (line_for_error (start_line), parse_error);
+	error_with_line (line_for_error (start_line), "%s", parse_error);
 	break;
       }
       i++;

--- a/gcc_arm/real.c
+++ b/gcc_arm/real.c
@@ -5703,8 +5703,6 @@ mtherr (name, code)
      char *name;
      int code;
 {
-  char errstr[80];
-
   /* The string passed by the calling program is supposed to be the
      name of the function in which the error occurred.
      The code argument selects which error message string will be printed.  */
@@ -5727,9 +5725,8 @@ mtherr (name, code)
     name = "modulus";
   else if (strcmp (name, "esqrt") == 0)
     name = "square root";
-  sprintf (errstr, "%s during real %s", ermsg[code], name);
   if (extra_warnings)
-    warning (errstr);
+    warning ("%s during real %s", ermsg[code], name);
   /* Set global error message word */
   merror = code + 1;
 }

--- a/gcc_arm/toplev.c
+++ b/gcc_arm/toplev.c
@@ -1393,7 +1393,7 @@ fatal_insn (message, insn)
      char *message;
      rtx insn;
 {
-  error (message);
+  error ("%s", message);
   debug_rtx (insn);
   if (asm_out_file)
     fflush (asm_out_file);

--- a/libc/Makefile
+++ b/libc/Makefile
@@ -15,7 +15,7 @@ export AR := $(PREFIX)ar
 export AS := $(PREFIX)as
 endif
 
-SHELL := /bin/bash -o pipefail
+SHELL := bash -o pipefail
 
 ASFLAGS := -mcpu=arm7tdmi
 

